profiler/src/ProfilerEngine/Datadog.Linux.ApiWrapper/functions_to_wrap.c

```diff
@@ -122,3 +120,34 @@ int dladdr(const void* addr_arg, Dl_info* info)
 
     return result;
 }
+
+#define INHIBIT_SIGNALS_FOR(fn_name)                                            \
+    /* Function pointers to hold the value of the musl libc functions */            \
+    static void (*__real_##fn_name)() = NULL;                                       \
+                                                                                    \
+    void fn_name()                                                                  \
+    {                                                                               \
+        if (__real_##fn_name == NULL)                                               \
+        {                                                                           \
+            __real_##fn_name = dlsym(RTLD_NEXT, #fn_name);                          \
```

Comment:
Maybe check that __real_##fn_name is different from nullptr
Would it be possible that a certain distrib does not contains the entry point we want to highjack?

Reply:
Good point, this is only for alpine

Reply:
but if it does not exist it will never be called

Reply:
✅ ok